### DISCUSSION TITLE
Console Warning: CANNOT USE API proposals

### DISCUSSION THREAD

**Discussion by asr1** [microsoft/vscode-discussions#887] [Extension Development QnA] [answered]
I have a VSCode extension, and when I look at the Window output, I see the following error:

```
[error] Extension 'office-script.office-script-extension CANNOT USE these API proposals 'ipc'. You MUST start in extension development mode or use the --enable-proposed-api command line flag
[info] Updating additional builtin extensions cache
```
Is this a known issue?

**Comment by gjsjohnmurray** [comment]
Are you the developer of the extension? If not, please contact them. The "Report Issue" from the "Help" menu in VS Code can help with this.

**Reply by asr1** [reply]
I am the developer, yes.

**Comment by gjsjohnmurray** [comment]
Have you reviewed the information at https://code.visualstudio.com/api/advanced-topics/using-proposed-api yet?

**Reply by asr1** [reply]
Yes, I added that parameter to my package.json, but continued to see the same error. I am not aware of a way to start the extension using the `--enable-proposed-api` command line flag, as the error message suggests.

**Reply by gjsjohnmurray** [reply]
Did you try this suggestion from that page?

> If you'd like to set it so that your extension using the proposed API is always available to use on every launch of VS Code Insiders, you can run the Preferences: Configure Runtime Arguments command to edit the .vscode-insiders/argv.json file to set a list of enabled extensions.

**Comment by r3m0t** [comment, marked as answer]
I am guessing you aren't using the `ipc` api intentionally? You didn't copy or download the below file into your project?

https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.ipc.d.ts#L35

This can happen if you are e.g. logging or using `JSON.stringify` on the `ExtensionContext` object. The attribute `messagePassingProtocol` would be read which causes this check and the associated warning. It's better to just log specific attributes of the object.

**Reply by asr1** [reply]
That makes sense, thank you!
